### PR TITLE
Search: Implemented showing the highlighted text in the search results

### PR DESCRIPTION
**Pull request by mikeger**
- [x] Implemented the search text field
- [x] Showing the count of results in the cell
- [x] Showing all the highlights
- [x]  Showing the word before the highlighted one
- [x] Tests

### CODE REVIEW

**Comment by mikeger** [comment]
@daehn

File: Wire-iOS/Sources/UserInterface/Collections/TextSearch/TextSearchResultCell.swift [hunk 1]
```diff
@@ -19,58 +19,37 @@
...
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
         
-        let ellipsisString = NSAttributedString(string: String.ellipsis, attributes: attributes)
-        return ellipsisString + self
+        self.layer.cornerRadius = ceil(self.bounds.height / 2.0)
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Collections/TextSearch/TextSearchResultCell.swift]: I think we should also update the `cornerRadius` initially, I remember us having issues with `layoutSubviews` not always being called in the `FileTransferCell`.

**Comment by mikeger** [comment]
@daehn please re-review

File: Wire-iOS/Sources/UserInterface/Collections/TextSearch/TextSearchResultCell.swift [hunk 2]
```diff
@@ -132,71 +132,29 @@ internal class SearchResultCountBadge: UIView {
...
+        self.messageTextLabel.query = self.query
+        self.messageTextLabel.resultText = text
         
-        let attributedText = NSMutableAttributedString(string: text, attributes: [NSFontAttributeName: font])
         let queryComponents = query.components(separatedBy: .whitespacesAndNewlines)
         
-        let currentRange = text.range(of: queryComponents,
-                                      options: [.diacriticInsensitive, .caseInsensitive],
-                                      range: text.startIndex..<text.endIndex)
+        let totalMatches = (text as NSString).allRanges(of: queryComponents, options: [.diacriticInsensitive, .caseInsensitive]).map { $1.count }.reduce(0, +)
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Collections/TextSearch/TextSearchResultCell.swift]: Maybe the `SearchResultsLabel` could expose this so we can avoid calling this multiple times (as I understand it the label calls this method internally as well).

File: Wire-iOS Tests/NSAttributedStringHighlightTests.swift [hunk 3]
```diff
@@ -0,0 +1,248 @@
+//
+//  NSAttributedStringHighlightTests.swift
+//  Wire-iOS
+//
+//  Created by Mihail Gerasimenko on 2/14/17.
```
Comment (daehn) [inline Wire-iOS Tests/NSAttributedStringHighlightTests.swift]: Missing copyright header

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m [hunk 4]
```diff
@@ -372,6 +372,14 @@ - (BOOL)definesPresentationContext
     return YES;
 }
 
+- (void)didReceiveMemoryWarning
+{
+    [super didReceiveMemoryWarning];
+    if (self.collectionController.view.window == nil) {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m]: 👍

**Comment by mikeger** [comment]
@daehn please merge if it's ok from your side

**Pull request merged**